src/app/providers/NFT/BoosterContext.tsx

```diff
@@ -0,0 +1,92 @@
+'use client'
+import { AVERAGE_BLOCKTIME, NFT_BOOSTER_DATA_URL } from '@/lib/constants'
+import { axiosInstance } from '@/lib/utils'
+import { useQuery } from '@tanstack/react-query'
+import { createContext, ReactNode, useContext, useMemo } from 'react'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+
+interface HolderRewards {
+  estimatedRBTCRewards: bigint
+  estimatedRIFRewards: bigint
+  boostedRBTCRewards: bigint
+  boostedRIFRewards: bigint
+}
+
+interface Holders {
+  [holderAddress: string]: HolderRewards
+}
+
+interface BoostData {
+  nftContractAddress: Address
+  boostPercentage: bigint
+  calculationBlock: bigint
+  holders: Holders
+}
+interface NFTBoosterContext {
+  boostData?: BoostData
+  isLoading?: boolean
+  error?: Error | null
+  currentBoost?: HolderRewards
+}
+
+const BoosterContext = createContext<NFTBoosterContext>({
+  boostData: {} as BoostData,
+})
+
+interface BoosterContextProviderProps {
+  children: ReactNode
+}
+export const BoosterProvider = ({ children }: BoosterContextProviderProps) => {
+  const { address } = useAccount()
+  const { data: boostData, isLoading, error } = useFetchBoostData()
+  const value = useMemo(() => {
+    if (!boostData && !isLoading && !error) {
+      return {}
+    }
+
+    if (!address) {
+      return { boostData, isLoading, error }
+    }
+
+    const currentBoost = boostData?.holders[address] || {
+      estimatedRBTCRewards: BigInt(0),
+      estimatedRIFRewards: BigInt(0),
+      boostedRBTCRewards: BigInt(0),
+      boostedRIFRewards: BigInt(0),
+    }
+
+    return { data: boostData, isLoading, error, currentBoost }
+  }, [boostData, address, isLoading, error])
+  return <BoosterContext.Provider value={value}>{children}</BoosterContext.Provider>
+}
+
+export const useNFTBoosterContext = () => useContext(BoosterContext)
+
+const useFetchBoostData = () => {
+  const {
+    data: latestFile,
+    isLoading: isFileDataLoading,
+    error: fileDataError,
+  } = useQuery<string>({
+    queryFn: () => axiosInstance.get(`${NFT_BOOSTER_DATA_URL}/latest`),
+    queryKey: ['nftBoosterLatestFile'],
+    refetchInterval: AVERAGE_BLOCKTIME,
+  })
+  const {
```

Comment:
we use the content of latest to understand if there is an active campaign or not.

Reply:
If the content is `None`, no active campaign.

Reply:
done